test/Multicall.t.sol

```diff
@@ -106,4 +107,112 @@ contract MulticallTest is Test {
         assertEq(multicall.msgValueDouble(), 200);
     }
+
+    // revert bubbling
+    function test_multicall_bubbleRevert_string() public {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] =
+            abi.encodeWithSelector(MockMulticall(multicall).functionThatRevertsWithString.selector, "errorString");
+
+        vm.expectRevert("errorString");
+        multicall.multicall(calls);
+    }
+
+    function test_multicall_bubbleRevert_4bytes() public {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeWithSelector(MockMulticall(multicall).revertWith4Bytes.selector);
+
+        // revert is caught
+        vm.expectRevert(MockMulticall.Error4Bytes.selector);
+        multicall.multicall(calls);
+
+        // confirm expected length of the revert
+        try multicall.revertWith4Bytes() {}
+        catch (bytes memory reason) {
+            assertEq(reason.length, 4);
+        }
+    }
+
+    function test_fuzz_multicall_bubbleRevert_36bytes(uint8 num) public {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeWithSelector(MockMulticall(multicall).revertWith36Bytes.selector, num);
+
+        // revert is caught
+        vm.expectRevert(abi.encodeWithSelector(MockMulticall.Error36Bytes.selector, num));
+        multicall.multicall(calls);
+
+        // confirm expected length of the revert
+        try multicall.revertWith36Bytes(num) {}
+        catch (bytes memory reason) {
+            assertEq(reason.length, 36);
+        }
+    }
+
+    function test_fuzz_multicall_bubbleRevert_68bytes(uint256 a, uint256 b) public {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeWithSelector(MockMulticall(multicall).revertWith68Bytes.selector, a, b);
+
+        // revert is caught
+        vm.expectRevert(abi.encodeWithSelector(MockMulticall.Error68Bytes.selector, a, b));
+        multicall.multicall(calls);
+
+        // confirm expected length of the revert
+        try multicall.revertWith68Bytes(a, b) {}
+        catch (bytes memory reason) {
+            assertEq(reason.length, 68);
+        }
+    }
+
+    function test_fuzz_multicall_bubbleRevert_arbitraryBytes(uint16 length) public {
+        length = uint16(bound(length, 0, 4096));
+        bytes memory data = new bytes(length);
+        for (uint256 i = 0; i < data.length; i++) {
+            data[i] = bytes1(uint8(i));
+        }
+
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeWithSelector(MockMulticall(multicall).revertWithBytes.selector, data);
+
+        // revert is caught
+        vm.expectRevert(abi.encodeWithSelector(MockMulticall.ErrorBytes.selector, data));
+        multicall.multicall(calls);
+
+        // confirm expected length of the revert
+        try multicall.revertWithBytes(data) {}
+        catch (bytes memory reason) {
+            // errors with 0 bytes are by default 64 bytes of data (length & pointer?) + 4 bytes of selector
```

Comment:
errors with 0 bytes of data outside the selector? thats super interesting.. didnt know that i guess that explains the 68 number